src/hooks/use-validation.jsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react';
+
+export default function useValidation(readOnly, helpText, inputRef) {
+  const [validationMessage, setValidationMessage] = useState('');
+
+  useEffect(() => {
+    if (!inputRef.current) return;
+
+    if (!inputRef.current.validity.valid && !inputRef.current.validity.customError) {
+      setValidationMessage(inputRef.current.validationMessage);
+      return;
+    }
+
+    if (!readOnly && helpText) {
+      inputRef.current.setCustomValidity(helpText);
+      setValidationMessage(helpText);
+    } else {
+      inputRef.current.setCustomValidity('');
+      setValidationMessage('');
+    }
+  });
```

Comment:
We should probably make this hook only run on changes to our function params:

```suggestion
  }, [readOnly, helpText, inputRef]);
```